packages/babel-helper-function-name/src/index.js

```diff
@@ -140,8 +140,7 @@ export default function ({ node, parent, scope, id }) {
     if (t.isIdentifier(id)) {
       const binding = scope.parent.getBinding(id.name);
       if (binding && binding.constant && scope.getBinding(id.name) === binding) {
-        // always going to reference this method
-        node.id = id;
+        node.id = Object.assign({}, id, { name: id.loc.identifierName });
```

Comment:
I think you can use object spread here (`{ ...id, name: id.loc.identifierName }`)